oauth2_http/java/com/google/auth/oauth2/AwsCredentials.java

```diff
@@ -137,18 +137,21 @@ private void validateMetadataServerUrls() {
       validateMetadataServerUrlIfAny(this.imdsv2SessionTokenUrl, "imdsv2_session_token_url");
     }
 
-    private static void validateMetadataServerUrlIfAny(String urlString, String nameInConfig) {
-      if (urlString != null) {
-        try {
-          URL url = new URL(urlString);
-          String host = url.getHost();
-          if (!host.equals("169.254.169.254") && !host.equals("[fd00:ec2::254]")) {
-            throw new IllegalArgumentException(
-                String.format("Invalid host %s for %s.", host, nameInConfig));
-          }
-        } catch (MalformedURLException malformedURLException) {
-          throw new IllegalArgumentException(malformedURLException);
+    @VisibleForTesting
+    static void validateMetadataServerUrlIfAny(String urlString, String nameInConfig) {
+      if (urlString == null || urlString.trim().length() == 0) {
+        return;
+      }
+
+      try {
+        URL url = new URL(urlString);
+        String host = url.getHost();
+        if (!host.equals("169.254.169.254") && !host.equals("[fd00:ec2::254]")) {
```

Comment:
can host have different casing?

Reply:
Apparently both lower and upper case are acceptable but as per RFC, they should be lower case https://www.rfc-editor.org/rfc/rfc5952#section-4.3